pkg/diskmaker/diskmaker.go

```diff
@@ -180,8 +192,27 @@ func (d *DiskMaker) symLinkDisks(diskConfig *DiskConfig) {
 
 	for storageClass, deviceArray := range deviceMap {
 		for _, deviceNameLocation := range deviceArray {
+			// get PV creation lock which checks for existing symlinks to this device
+			pvLock, pvLocked, existingSymlinks, err := internal.GetPVCreationLock(
+				deviceNameLocation.diskNamePath,
+				d.symlinkLocation,
+			)
+			unlockFunc := func() {
+				err := pvLock.Unlock()
+				if err != nil {
+					klog.Errorf("failed to unlock device: %+v", err)
+				}
+			}
+			defer unlockFunc()
+			if len(existingSymlinks) > 0 { // already claimed, fail silently
+				continue
+			} else if err != nil || !pvLocked { // locking failed for some other reasion
+				klog.Errorf("not symlinking, could not get lock: %v", err)
+				continue
```

Comment:
We should report errors as events in case we can't claim a device because it was previously symlinked.

Reply:
added